### PR TITLE
clustering: wrapper for Scipy's hiearchical clustering

### PR DESCRIPTION
**Pull request by glouppe**
This PR implements a scikit-learn compatible wrapper for Scipy's hiearchical clustering.

* [x] Implementation
* [x] Tests
* [x] Docstrings

### CODE REVIEW

**Comment by glouppe** [comment]
Ready for reviews :)

CC: @MSusik @etzemis @petros-ioannidis

File: beard/clustering/wrappers.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+        return self
+
+    @property
+    def labels_(self):
+        """Compute the labels assigned to the input data.
+
+        Note that labels are computed on-the-fly from the linkage matrix,
+        based on the value of self.threshold or self.n_clusters.
+        """
```
Comment (MSusik) [inline beard/clustering/wrappers.py]: Maybe the result should me memorized, and if any of `threshold` or `n_clusters` changes, you can wipe out the remembered result.

Not sure if it necessary/useful, though.

Comment (glouppe) [inline beard/clustering/wrappers.py]: I see your point, but I would rather not complexify this for the time being. If this proves to be a bottleneck, then we will improve it later.

Comment (MSusik) [inline beard/clustering/wrappers.py]: OK, then I approve merging it.

**Pull request merged**